lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -1153,14 +1153,32 @@ def install_puppet_agent_dev_repo_on( hosts, opts )
               if arch == 'x86_64'
                 arch = 'i386'
               end
+              release_path << "solaris/#{version}/#{opts[:puppet_collection]}"
+              solaris_revision_conjunction = '-'
+              revision = '1'
               if version == '10'
                 # Solaris 10 uses / as the root user directory. Solaris 11 uses /root.
                 onhost_copy_base = '/'
+                solaris_release_version = ''
+                pkg_suffix = 'pkg.gz'
+                solaris_name_conjunction = '-'
+                component_version = opts[:puppet_agent_version]
+              elsif version == '11'
+                # Ref:
+                # http://www.oracle.com/technetwork/articles/servers-storage-admin/ips-package-versioning-2232906.html
+                solaris_release_version = ',5.11' # injecting comma to prevent from adding another var
+                pkg_suffix = 'p5p'
+                solaris_name_conjunction = '@'
+                component_version = opts[:puppet_agent_version]
+                component_version.gsub!(/[a-zA-Z]/, '')
+                component_version.gsub!(/(^-)|(-$)/, '')
```

Comment:
@johnduarte did you see the previous comments about using `component_version = component_version.gsub` instead of `component_version.gsub!`? `gsub!` returns nil when no substitutions were made, which probably isn't what you want here.